src/main/java/com/fiap/techChallenge/application/product/ProductServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.fiap.techChallenge.application.product;
+
+import com.fiap.techChallenge.domain.Product;
+import com.fiap.techChallenge.infrastructure.product.ProductEntity;
+import com.fiap.techChallenge.infrastructure.product.ProductRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class ProductServiceImpl implements ProductService {
+
+    private ProductRepository productRepository;
+
+    @Override
+    public Product saveProduct(Product product) {
+        ProductEntity productEntity = ProductEntity.builder()
+                .price(product.getPrice())
+                .category(product.getCategory())
+                .description(product.getDescription())
+                .images(product.getImages())
+                .build();
```

Comment:
O que acha de criarmos um método dentro da classe Product que ja devolve a entidade criada pra gente? ex:
`public ProductEntity toEntity() {
return ProductEntity.builder()
                .price(this.getPrice())
                .category(this.getCategory())
                .description(this.getDescription())
                .images(this.getImages())
                .build();
}
`

Reply:
Boa, fiz o ajuste